src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java

```diff
@@ -0,0 +1,268 @@
+package com.pubnub.api.models.consumer.push.payload;
+
+import com.pubnub.api.enums.PNPushEnvironment;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Setter
+@Accessors(chain = true)
+public class PushPayloadHelper {
+
+    private Map<String, Object> commonPayload;
+    private FCMPayload fcmPayload;
+    private MPNSPayload mpnsPayload;
+    private APNSPayload apnsPayload;
+
+    public Map<String, Object> build() {
+        Map<String, Object> payload = new HashMap<>();
+
+        if (apnsPayload != null) {
+            Map<String, Object> apnsMap = apnsPayload.toMap();
+            if (!apnsMap.isEmpty()) {
+                payload.put("pn_apns", apnsMap);
```

Comment:
Should we separate APNS vs APNS2?

Reply:
@davidnub 

Actually it's separated already but the naming isn't that straightforward, which is now improved in another PR #193